airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbStateIterator.java

```diff
@@ -46,32 +52,42 @@ class MongoDbStateIterator implements Iterator<AirbyteMessage> {
   /**
    * Pointer to the last document _id seen by this iterator, necessary to track for state messages.
    */
-  private String lastId = null;
+  private Object lastId;
 
   /**
    * This iterator outputs a final state when the wrapped `iter` has concluded. When this is true, the
    * final message will be returned.
    */
   private boolean finalStateNext = false;
 
+  /**
+   * Tracks if the underlying iterator threw an exception. This helps to determine the final state
+   * status emitted from the final next call.
+   */
+  private boolean iterThrewException = false;
+
   /**
    * Constructor.
    *
    * @param iter {@link MongoCursor} that iterates over Mongo documents
    * @param stateManager {@link MongoDbStateManager} that manages global and per-stream state
    * @param stream the stream that this iterator represents
    * @param emittedAt when this iterator was started
-   * @param checkpointInterval how often a state message should be emitted.
+   * @param checkpointInterval how often a state message should be emitted based on number of
+   *        messages.
+   * @param checkpointDuration how often a state message should be emitted based on time.
    */
-  MongoDbStateIterator(final MongoCursor<Document> iter,
-                       final MongoDbStateManager stateManager,
-                       final ConfiguredAirbyteStream stream,
-                       final Instant emittedAt,
-                       final int checkpointInterval) {
+  public MongoDbStateIterator(final MongoCursor<Document> iter,
```

Comment:
I dont think this class is adding the metadata to the records like we do for debezium columns. this needs to use the class `MongoDbCdcConnectorMetadataInjector` and add the metadata like the cursor value. The method `addMetaDataToRowsFetchedOutsideDebezium` needs to be used

Reply:
@subodh1810 I took a shot at this.  Please take a look and see if I got it correct.